common/notify_stream/src/lib.rs

```diff
@@ -140,37 +146,79 @@ impl Watcher {
         let rx = Rc::clone(&self.rx);
         Box::pin(stream::unfold(rx, |rx| async move {
             loop {
-                let received = rx.recv().await.expect("channel can not be closed");
+                let received = rx.recv().await.expect("channel can not be closed").unwrap();
                 log::trace!("received raw notify event: {:?}", received);
-                if let Some(mapped_event) = match received {
-                    DebouncedEvent::NoticeRemove(p) => Some(Event::Remove(p)),
-                    DebouncedEvent::Create(p) => Some(Event::Create(p)),
-                    DebouncedEvent::Write(p) => Some(Event::Write(p)),
-                    DebouncedEvent::Rename(source, dest) => Some(Event::Rename(source, dest)),
-                    // TODO: Define what to do with Rescan
-                    DebouncedEvent::Rescan => Some(Event::Rescan),
-                    DebouncedEvent::Error(e, p) => Some(Event::Error(e.into(), p)),
-                    // NoticeWrite can be useful but we don't use it
-                    DebouncedEvent::NoticeWrite(_) => None,
-                    // Ignore `Remove`: we use `NoticeRemove` that comes before in the flow
-                    DebouncedEvent::Remove(_) => None,
-                    // Ignore attribute changes
-                    DebouncedEvent::Chmod(_) => None,
+                if let Some(mapped_event) = match received.kind {
```

Comment:
The `receive()` method where the raw filesystem events are mapped. The structure of these events has changed so as results the mapping are different.